exporter/exporterhelper/internal/queue_sender.go

```diff
@@ -46,12 +48,34 @@ type QueueConfig struct {
 	NumConsumers int `mapstructure:"num_consumers"`
 	// QueueSize is the maximum number of requests allowed in queue at any given time.
 	QueueSize int `mapstructure:"queue_size"`
-	// Blocking controls the queue behavior when full.
-	// If true it blocks until enough space to add the new request to the queue.
+
+	// Deprecated: [v0.123.0] use `block_on_overflow`.
 	Blocking bool `mapstructure:"blocking"`
+
+	// BlockOnOverflow determines the behavior when the component's TotalSize limit is reached.
+	// If true, the component will wait for space; otherwise, operations will immediately return a retryable error.
+	BlockOnOverflow bool `mapstructure:"block_on_overflow"`
```

Comment:
@jmacd @dmitryax since this is not released yet anyway, I was thinking of a potentially better name: "DropWhenFull"? Would this clarify what happens for the final user? WDYT?

Reply:
But we don't drop, we reject the data. The sender (an SDK) may potentially retry

Reply:
From this component perspective we drop correct? Happy to hear other options, but I heard internal feedback that is not clear what happens if `block_on_overflow` is `false` from the name.

Reply:
> From this component perspective we drop correct?

I don't think so. I'd use term "drop" if it happens after the async queue, not before

Reply:
Until we resolve https://github.com/open-telemetry/opentelemetry-collector/issues/12393, `ErrQueueIsFull` is still retriable. I believe it should be kept that way even when change the errors to non-retriable by default. 

I think `reject_when_full` or `reject_on_overflow` would work as well. However `block_on_overflow` seems better to me as we can keep `false` to be default value.

Reply:
Let's keep it as is.